spec.html

```diff
@@ -88,18 +88,24 @@ <h1>Runtime Semantics: Evaluation</h1>
           1. IfAbruptRejectPromise(_specifierString_, _promiseCapability_).
           1. <ins>Let _argRef_ be the result of evaluating the second |AssignmentExpression|.</ins>
           1. <ins>Let _arg_ be ? GetValue(_argRef_).</ins>
-          1. <ins>If _arg_ is *undefined*, let _assertions_ be an empty List.</ins>
-          1. <ins>Otherwise,</ins>
-            1. <ins>Let _supportedAssertions_ ! HostGetSupportedImportAssertions().</ins>
+          1. <ins>Let _assertions_ be a new empty List.</ins>
+          1. <ins>If _arg_ is not *undefined*,</ins>
+            1. <ins>If Type(_arg_) is not Object, then</ins>
+              1. <ins>Perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; a newly created *TypeError* object &raquo;).</ins>
```

Comment:
another alternative would be using `ToObject`, and only rejecting on `null`.

Reply:
Note, I suggested the equivalent of this in https://github.com/tc39/proposal-record-tuple/issues/107 and got a bunch of skepticism as a response. From that thread and looking through the JS stdlib, I came to the conclusion that we have plenty of precedent to stick with checking "Type(_arg_) is Object" (potentially later expanded to "Object or Record or Tuple") for cases like this. Note also the discussion in https://github.com/tc39/ecma402/issues/480 where I guess we did go to with ToObject for options bags.

Reply:
This was discussed in April's meeting and it seems like there was pretty broad agreement that throwing for non-Object, rather than converting `ToObject`, is the direction we want to go. This matches what ECMA 402 is doing with GetOption() ([link](https://github.com/tc39/notes/blob/master/meetings/2020-09/sept-22.md#getoption-in-ecma-262)).

@bmeck you mentioned that you had varied opinions on whether to reject or not based on whether we pull off the prototype (or something along those lines -- I didn't quite follow at the time). You said that you wanted to follow up offline, so I wanted to give you a chance to chime in here in case there was any concern you wanted to raise before I go ahead and land this PR.